crates/core/src/sql/compiler.rs

```diff
@@ -371,15 +374,17 @@ mod tests {
 
     #[test]
     fn compile_index_eq() -> ResultTest<()> {
-        let (db, _) = make_test_db()?;
-        let mut tx = db.begin_tx();
+        let (db, _tmp) = make_test_db()?;
```

Comment:
Any reason for adding `_tmp`?

Reply:
`make_test_db()` creates a temporary dir (for db) which lives until its binding is in the scope.
`db.commit_tx()` below writes to this temporary dir, so `_tmp` is require to bind dir scope to current test.
Why this was not a problem before? because we were using the single `tx` in all the tests and never writting anything back to disk.  Now, we need to swtich between type of tx.

Reply:
Note, there is a semantic difference between `_` and `_tmp` for lifetimes